apps/web/frontend/components/projectInvitation.tsx

```diff
@@ -0,0 +1,85 @@
+import { useRef } from 'react'
+import { BiCopyAlt } from 'react-icons/bi'
+import { useMutation } from 'urql'
+
+import { toastSuccess } from '@progwise/timebook-ui'
+
+import { graphql } from '../generated/gql'
+
+export const ProjectInvitationFragment = graphql(`
+  fragment ProjectInvitationFragment on Project {
+    id
+  }
+`)
+
+const projectMembershipInvitationMutation = graphql(`
+  mutation projectMembershipInvitation($projectId: ID!) {
+    projectMembershipInvitationCreate(projectId: $projectId) {
+      id
+      invitationKey
+      expireDate
+    }
+  }
+`)
+
+interface ProjectInvitationProps {
+  projectId: string
+}
+
+export const ProjectInvitation = ({ projectId }: ProjectInvitationProps) => {
+  const [{ data, fetching }, invitationKeyCreate] = useMutation(projectMembershipInvitationMutation)
+
+  const dialogReference = useRef<HTMLDialogElement>(null)
+
+  const invitationLink = `${process.env.NEXT_PUBLIC_APP_URL}/projects/join/${data?.projectMembershipInvitationCreate.invitationKey}`
+
+  return (
+    <>
+      <button
+        className="btn btn-outline btn-sm"
+        type="button"
+        onClick={() => {
+          dialogReference.current?.showModal()
+          invitationKeyCreate({ projectId })
+        }}
+      >
+        Invite
+      </button>
+      <dialog className="modal" ref={dialogReference}>
+        <div className="modal-box whitespace-normal text-base-content">
+          <h3 className="mb-4 text-lg font-bold">Invitation key</h3>
+          <div className="text-base">
+            <p>Here is the invitation key:</p>
+            <div className="my-2 flex items-center">
+              <input
+                className="input input-bordered grow bg-neutral py-2 text-neutral-content"
+                value={invitationLink}
+                readOnly
+              />
+              <button
+                className="btn btn-circle btn-ghost gap-2 text-xl"
+                onClick={() => {
+                  navigator.clipboard.writeText(invitationLink ?? '')
```

Comment:
not needed, `invitationLink` is always defined
```suggestion
                  navigator.clipboard.writeText(invitationLink)
```